pkg/lib/DynamicListForm.jsx

```diff
@@ -65,6 +70,16 @@ export class DynamicListForm extends React.Component {
                                     additem: this.addItem,
                                     options: this.props.options,
                                     itemCount: Object.keys(dialogValues.list).length,
+                                    validationFailed: validationFailed && validationFailed[idx],
+                                    onValidationChange: value => {
+                                        const delta = validationFailed ? [...validationFailed] : [];
+                                        delta[idx] = value;
+
+                                        if (Object.keys(delta[idx]).length == 0)
+                                            delete delta[idx];
```

Comment:
Can you please add a comment what this code does?

This back-and-forth on `delta[idx]` also looks weird. Perhaps rewrite this to 
```js
if (Object.keys(value).length > 0)
    delta[idx] = value;
```
or similar? But I still don't know exactly what/why this happens, in abstract English.